src/Affix.ts

```diff
@@ -0,0 +1,15 @@
+import {Stat} from './Stat'
+
+interface IAffix {
+  name: string
+  affixType: string
+  stats: Stat
+}
+
+export class Affix implements IAffix {
```

Comment:
I would probably defer doing equipment, effects, and affixes until you've got more functionality tested and wired up, so that you don't have to change as many classes if you have to change the patterns you use.